### PR TITLE
Fading collapsible

### LINKED ISSUE #148: Add Fading'Accordion'
Add expandable container with fading effect usable in Mdx files to collapse lengthy secondary content.
Use it on /components installation section
Take inspo from TabbedCode colllapsible section

### PR DESCRIPTION
**Pull request by ChcJohnie**
closes #148 

Adds FadingCollapsible wrapping container which hides overflowing content unless expanded.
Used in Doc variant with defaults, utilized by /components installation block.
Also reused in TabbedCode component.

Note: as docs variant was not containing visually distinctive block (like with code), it feeled bit strange for me. I tried to mitigate with left side subtle border, so structure is more defined but Im defo open to other design suggestions.

### CODE REVIEW

File: src/docs/components/Code/TabbedCode.tsx [hunk 1]
```diff
@@ -17,100 +21,62 @@ export interface TabbedCodeItem {
...
-
 export const TabbedCode = ({
   collapsedClassName,
-  expandedClassName,
   className,
   tabs,
 }: TabbedCodeProps) => {
-  const isExpandable = collapsedClassName || expandedClassName;
+  const isExpandable = typeof collapsedClassName === "string";
+  const Wrapper = isExpandable ? FadingCollapsible : React.Fragment;
```
Comment (ChcJohnie) [inline src/docs/components/Code/TabbedCode.tsx:33]: This Wrapper related code is bit Vueish way of writing related logic. Is it okay in React?

Comment (iamhectorsosa) [reply src/docs/components/Code/TabbedCode.tsx:33]: Yes, this is also a valid pattern in React.

File: src/docs/components/InstallationDocs/DocCollapsible.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import {
+  FadingCollapsible,
+  FadingCollapsibleProps,
+} from "@components/FadingCollapsible";
+
+const defaultOptions: Omit<FadingCollapsibleProps, "children"> = {
+  expandButtonEl: "Show more",
+  collapseButtonEl: "Hide section",
+  className: "space-y-8 md:space-y-12 border-l border-stone-700 pl-2 md:pl-4",
```
Comment (iamhectorsosa) [inline src/docs/components/InstallationDocs/DocCollapsible.tsx:9]: This border left looks a bit off. I'd remove it and just leave the background gradient.

**Review by iamhectorsosa — COMMENTED** [review]
@ChcJohnie this is exactly what I was looking for but we just need to hone some of the implementation details. Nice work!

**Comment by iamhectorsosa** [comment]
I wrote a comment that wasn't included in the review. Please ping me for a quick call to discuss this. @ChcJohnie

**Comment by ChcJohnie** [comment]
Applied some refactors discussed during call:
- renamed to CollapsibleContent
- refactored props (keeped same number but simplified naming/usage)
- adjusted in docs border (but keep free to refuse that 🙂 )

**Pull request merged**